packages/trader/src/Stores/Modules/Contract/contract-store.js

```diff
@@ -92,9 +102,25 @@ export default class ContractStore {
                     line_style: BARRIER_LINE_STYLES.SOLID,
                     not_draggable: true,
                 });
+                main_barrier.shade = getBarrierShade(contract_info);
+                main_barrier.shadeColor = get_color({
+                    status,
+                    profit,
+                    is_dark_mode: this.root_store.ui.is_dark_mode_on,
+                });
 
                 main_barrier.updateBarrierShade(true, contract_type);
                 result = [main_barrier];
+
+                if (/RESET(CALL|PUT)/.test(contract_type)) {
```

Comment:
please use a helper function for this check 🙏

Reply:
@kelly-binary Do we need a helper function for this? It's a one liner check, feels redundant to create a wrapper just for a regex test. Negar told me that we have this on binary static but no one seems to use it because everyone just writes their own one liner regex test, thus we should just use these regex tests instead of a function.

Reply:
I'd prefer to have a helper function although it is a simple regex test cos the idea is to avoid repetition if the same regex test is used in multiple places

Reply:
@kelly-binary That is what I was saying is redundant. Instead of `/RESET(CALL|PUT)/.test(contract_type)` we repeat `isResetCallPut(contract_type)`, the repetition is still there because the code only does one thing.

Reply:
@brandon-binary Since we're following the DRY principle in the styleguide. It's better to use a single helper function that does this for us, rather than having to write the regex test `/RESET(CALL|PUT)/.test(contract_type)` everytime we need to use it. This way whenever we need to change the test, we only need to change it in one file (i.e. the helper files)

Reply:
@kelly-binary Not necessarily. Since the test is so specific, if we change it, we also need to change the function name (it will be misleading otherwise), which in turn result in needing to change the function names elsewhere such as when importing and using.

Again, I don't think its worth it to put it in a function, its small and only does one specific thing. It would be as if we put `a + b` into a function `add(a, b)`, and needing to import it every time before use instead of using `a + b` directly.